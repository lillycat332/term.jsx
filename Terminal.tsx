import React from 'react';
import Box from './Box';
import './buttons.css';

type TerminalProps = {
	children: React.ReactNode
	style?: React.CSSProperties
	className?: string
	windowDecoration?: boolean
}

const macosWindowDecoration = () => {
	return (
		<>
			<div className='base-circle close-circle'></div>
			<div className='base-circle min-circle'></div>
			<div className='base-circle max-circle'></div>
		</>
	)
}

const Terminal: React.FunctionComponent<TerminalProps> = (props: TerminalProps) => {
	const { children, style, className, windowDecoration } = props;

	return (
		<div className={className ?? "code"}>
			<Box style={style}>
				{windowDecoration && macosWindowDecoration()}
				{children}
			</Box>
		</div >
	);
}

export default Terminal;