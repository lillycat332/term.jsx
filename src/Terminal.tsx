import React from 'react';
import Box from './Box';
import styles from './buttons.module.css';

export type TerminalProps = {
	children?: React.ReactNode
	style?: React.CSSProperties
	className?: string
	windowDecoration?: boolean
	title?: string
}

const defaultStyle = {
	borderRadius: 8,
	minHeight: "40vh",
	padding: 12,
	minWidth: "50vw",
	marginTop: "5%",
	boxShadow: `
  0 1px 1px rgba(0,0,0,0.11), 
	0 2px 2px rgba(0,0,0,0.11), 
	0 4px 4px rgba(0,0,0,0.11), 
	0 8px 8px rgba(0,0,0,0.11), 
	0 16px 16px rgba(0,0,0,0.11), 
	0 32px 32px rgba(0,0,0,0.11)`,
}

// const macosWindowDecoration = () => {
// 	return (
// 		<div className={styles['button-padding']}>
// 			<div className={[styles['base-circle'], styles['close-circle']].join(' ')}></div>
// 			<div className={[styles['base-circle'], styles['min-circle']].join(' ')}></div>
// 			<div className={[styles['base-circle'], styles['max-circle']].join(' ')}></div>
// 		</div>
// 	)
// }

export function Terminal(props: TerminalProps) {
	const { children, style, className, windowDecoration } = props;

	const boxStyle = {
		...defaultStyle,
		...style
	}

	return (
		<div className={className ?? "code"}>
			<Box style={boxStyle}>
				{windowDecoration && (
					<div className={[styles['button-padding']].join(' ')}>
						<div className={[styles['base-circle'], styles['close-circle']].join(' ')}></div>
						<div className={[styles['base-circle'], styles['min-circle']].join(' ')}></div>
						<div className={[styles['base-circle'], styles['max-circle']].join(' ')}></div>
					</div>
				)} {children}
			</Box >
		</div >
	);
}
