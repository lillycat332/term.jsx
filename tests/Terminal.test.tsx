/**
 * @jest-environment jsdom
 */

import { render } from '@testing-library/react';

import { Terminal } from '../src/Terminal'


const children = (
	<div>
		<p id="child">this is a child</p>
	</div>
)

describe('Terminal', () => {
	it('renders', () => {
		const { container } = render(<Terminal />)
		expect(container).toMatchSnapshot()
	});

	it('renders with window decoration', () => {
		const { container } = render(<Terminal windowDecoration={true} />)
		expect(container).toMatchSnapshot()
		expect(container.querySelector('.base-circle')).not.toBeNull()
	});

	it('renders with children', () => {
		const { container } = render(<Terminal>{children}</Terminal>)
		expect(container).toMatchSnapshot()
		expect(container.querySelector('#child')).toBeTruthy()
	})

	it('renders with window decoration and children', () => {
		const { container } = render(<Terminal windowDecoration={true}>{children}</Terminal>)
		expect(container).toMatchSnapshot()
		expect(container.querySelector('#child')).toBeTruthy()
		expect(container.querySelector('.base-circle')).not.toBeNull()
	})
})
