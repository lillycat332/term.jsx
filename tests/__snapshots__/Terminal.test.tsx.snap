// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Terminal renders 1`] = `
<div>
  <div
    class="code"
  >
    <div>
       
    </div>
  </div>
</div>
`;

exports[`Terminal renders with children 1`] = `
<div>
  <div
    class="code"
  >
    <div>
       
      <div>
        <p
          id="child"
        >
          this is a child
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Terminal renders with window decoration 1`] = `
<div>
  <div
    class="code"
  >
    <div>
       
      <div
        class="base-circle close-circle"
      />
      <div
        class="base-circle min-circle"
      />
      <div
        class="base-circle max-circle"
      />
    </div>
  </div>
</div>
`;

exports[`Terminal renders with window decoration and children 1`] = `
<div>
  <div
    class="code"
  >
    <div>
       
      <div
        class="base-circle close-circle"
      />
      <div
        class="base-circle min-circle"
      />
      <div
        class="base-circle max-circle"
      />
      <div>
        <p
          id="child"
        >
          this is a child
        </p>
      </div>
    </div>
  </div>
</div>
`;
